### DISCUSSION TITLE
Interrogate will be fallen back to cpu. Because DirectML device does not support it.

### DISCUSSION THREAD

**Discussion by Nhiira** [lshqqytiger/stable-diffusion-webui-directml#15] [General]
Followed every step having this issue with RX 6600 XT (driver 23.2.2), Windows 11.

venv "C:\stable-diffusion-webui-directml-master\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Commit hash: <none>
Installing requirements for Web UI
Launching Web UI with arguments:
Interrogate will be fallen back to cpu. Because DirectML device does not support it.
Warning: caught exception 'Torch not compiled with CUDA enabled', memory monitor disabled
No module 'xformers'. Proceeding without it.
Loading weights [fe4efff1e1] from C:\stable-diffusion-webui-directml-master\models\Stable-diffusion\sd-v1-4.ckpt
Creating model from config: C:\stable-diffusion-webui-directml-master\configs\v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Applying cross attention optimization (InvokeAI).
Textual inversion embeddings loaded(0):
Model loaded in 4.5s (load weights from disk: 1.2s, create model: 0.5s, apply weights to model: 0.6s, apply half(): 0.6s, move model to device: 1.6s).
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.

**Comment by Enferlain** [comment]
6800xt same

**Reply by Nhiira** [reply]
I made it work finally. So in short "Interrogate will be fallen back to cpu. Because DirectML device does not support it." is not a problem. it uses our GPU, that warning doesn't mean it'll use CPU while generating image.

But it's better start with "set COMMANDLINE_ARGS=--opt-sub-quad-attention --medvram --disable-nan-check --no-half --precision full" command in webui-user bat file

**Reply by Miraihi** [reply]
"Interrogate" is the feature in img2img that allows you to get prompt out of a picture. That has nothing to do with the image generation.

**Reply by Enferlain** [reply]
Alright. Just thought it wasn't working cuz it was slow af but I guess it's not an a100

**Comment by badconduct** [comment]
Having the same issue. 6800 XT. 
Webui just runs on the CPU, doesn't touch the GPU at all.
set COMMANDLINE_ARGS=--skip-torch-cuda-test --opt-sub-quad-attention --medvram --disable-nan-check --no-half --precision full

**Comment by Miraihi** [comment]
UPDATE: Danbooru interrogation works in the current commit ([265d626](https://github.com/lshqqytiger/stable-diffusion-webui-directml/commit/265d626471eacd617321bdb51e50e4b87a7ca82e)). CLIP interrogation still doesn't.
